### PR TITLE
 FINERACT-1450-fix

### PR DESCRIPTION
**Pull request by fynmanoj**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanRescheduleWithAdvancePaymentTest.java
Comment (avikganguly01) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanRescheduleWithAdvancePaymentTest.java:254]: LGTM

**Review by avikganguly01 — APPROVED** [review]

**Comment by vidakovic** [comment]
Hi... can we hold this one for a moment and not merge it until #2187 and #2188 are approved and merged? I think there won't be too much of an impact to your PR if we take the other two first, but those PRs were a bit painful to get right and would be important for cleanup. The PRs are ready, someone might approve this weekend, at the latest Monday. Thanks in advance for the patience.

**Comment by avikganguly01** [comment]
Both #2187 and #2188 seems to be merged. Can I go ahead and merge this now?

**Comment by vidakovic** [comment]
Currently investigating failing tests on develop... even minor lib updates from the dependency bot are failing... could this be related? If you have a moment to checkout current develop, run one of the failing tests and see if you can recognize something? Doing the same here. Thanks again.

**Comment by vidakovic** [comment]
```
testLoanRefundByCashCashBasedAccounting()
```
... this is the stacktrace I get:

```
2022-03-29 07:49:11.896 ERROR 194527 --- [pool-5-thread-2] a.f.p.s.s.SavingsSchedularInterestPoster : Interest posting job for savings failed for account 10

java.lang.NullPointerException: Cannot invoke "java.math.BigDecimal.divide(java.math.BigDecimal, java.math.MathContext)" because the return value of "org.apache.fineract.portfolio.savings.data.SavingsAccountData.getNominalAnnualInterestRateOverdraft()" is null
	at org.apache.fineract.portfolio.savings.service.SavingsAccountInterestPostingServiceImpl.getEffectiveOverdraftInterestRateAsFraction(SavingsAccountInterestPostingServiceImpl.java:364)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountInterestPostingServiceImpl.calculateInterestUsing(SavingsAccountInterestPostingServiceImpl.java:279)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountInterestPostingServiceImpl.postInterest(SavingsAccountInterestPostingServiceImpl.java:84)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountWritePlatformServiceJpaRepositoryImpl.postInterest(SavingsAccountWritePlatformServiceJpaRepositoryImpl.java:662)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountWritePlatformServiceJpaRepositoryImpl$$FastClassBySpringCGLIB$$fa18015c.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:698)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountWritePlatformServiceJpaRepositoryImpl$$EnhancerBySpringCGLIB$$8dd5f997.postInterest(<generated>)
	at org.apache.fineract.portfolio.savings.service.SavingsSchedularInterestPoster.call(SavingsSchedularInterestPoster.java:153)
	at org.apache.fineract.portfolio.savings.service.SavingsSchedularInterestPoster.call(SavingsSchedularInterestPoster.java:61)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

2022-03-29 07:49:11.896 ERROR 194527 --- [pool-5-thread-3] a.f.p.s.s.SavingsSchedularInterestPoster : Interest posting job for savings failed for account 12

java.lang.NullPointerException: Cannot invoke "java.math.BigDecimal.divide(java.math.BigDecimal, java.math.MathContext)" because the return value of "org.apache.fineract.portfolio.savings.data.SavingsAccountData.getNominalAnnualInterestRateOverdraft()" is null
	at org.apache.fineract.portfolio.savings.service.SavingsAccountInterestPostingServiceImpl.getEffectiveOverdraftInterestRateAsFraction(SavingsAccountInterestPostingServiceImpl.java:364)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountInterestPostingServiceImpl.calculateInterestUsing(SavingsAccountInterestPostingServiceImpl.java:279)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountInterestPostingServiceImpl.postInterest(SavingsAccountInterestPostingServiceImpl.java:84)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountWritePlatformServiceJpaRepositoryImpl.postInterest(SavingsAccountWritePlatformServiceJpaRepositoryImpl.java:662)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountWritePlatformServiceJpaRepositoryImpl$$FastClassBySpringCGLIB$$fa18015c.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:698)
	at org.apache.fineract.portfolio.savings.service.SavingsAccountWritePlatformServiceJpaRepositoryImpl$$EnhancerBySpringCGLIB$$8dd5f997.postInterest(<generated>)
	at org.apache.fineract.portfolio.savings.service.SavingsSchedularInterestPoster.call(SavingsSchedularInterestPoster.java:153)
	at org.apache.fineract.portfolio.savings.service.SavingsSchedularInterestPoster.call(SavingsSchedularInterestPoster.java:61)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
```

... is failing... just in case you have an idea.

**Comment by fynmanoj** [comment]
Sure, I shall check the test case

**Comment by vidakovic** [comment]
See this PR to easily identify where things are breaking: https://github.com/apache/fineract/pull/2212/files.
I have annotated the stuff with "// TODO: @fynmanoj, @avikganguly01... could you help find out why this is failing?" so that you can find it easier.

**Comment by vidakovic** [comment]
For your convenience I've also extracted the stacktraces that i could find... maybe that helps... it seems that something with JPA parameter passing for queries is off (missing values?). There are some EclipseLink warnings in the logs (I'll send on the mailing list).

**Comment by vidakovic** [comment]
@avikganguly01 @fynmanoj was not the issue... sorry for that... must be another dependency issue of some sort. If you find something suspicious though please let me know.

**Pull request merged**
